services/wallet/async/async_test.go

```diff
@@ -0,0 +1,59 @@
+package async
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAtomicGroupTerminatesOnOneCommandFailed(t *testing.T) {
+	ctx := context.Background()
+	group := NewAtomicGroup(ctx)
+
+	err := errors.New("error")
+	group.Add(func(ctx context.Context) error {
+		time.Sleep(1 * time.Millisecond)
```

Comment:
What do you think about removing the sleep (not to make the test take unnecessarily) but adding instead this failure command after the next command instead, ensuring the other is failing if finished after the `1ms` timeout?